TestResultSummaryService/parsers/BenchmarkParser.js

```diff
@@ -90,45 +91,40 @@ class BenchmarkParser extends Parser {
-                    curSearchString = curItr.value;
+                for ( let i = 0; i < curBench["metrics"].length; i++ ) {
 
+                    curMetric = curBench["metrics"][i].name;
+                    curSearchString = curItr.value;
                     // metric values will be found within the result of the outer regex
-                    if (BenchmarkMetric[curBenchmarkNamedRouted]["metrics"][i].outerRegex !== undefined) {
-                        if ( ( curRegexResult = BenchmarkMetric[curBenchmarkNamedRouted]["metrics"][i].outerRegex.exec( curSearchString ) ) !== null ) {
+                    if (curBench["metrics"][i].outerRegex !== undefined) {
+                        if ( ( curRegexResult = curBench["metrics"][i].outerRegex.exec( curSearchString ) ) !== null ) {
                             curSearchString = curRegexResult[1];
                         } else {
                             // No result for outer regex, skip this metric
                             continue;
                         }
                     }
-
-                    // multiple metric values are present
-                    if (BenchmarkMetric[curBenchmarkNamedRouted]["metrics"][i].regexRepeat === true) {
-                        curMetricValue = [];
-                        curRegexResult = BenchmarkMetric[curBenchmarkNamedRouted]["metrics"][i].regex.exec( curSearchString )
-
-                        while (curRegexResult !== null ) {
-                            curMetricValue.push(parseFloat(curRegexResult[1]));
-                            curRegexResult = BenchmarkMetric[curBenchmarkNamedRouted]["metrics"][i].regex.exec( curSearchString );
-                        }
-
-                    // single metric value
-                    } else {
-                        if ( ( curRegexResult = BenchmarkMetric[curBenchmarkNamedRouted]["metrics"][i].regex.exec( curSearchString ) ) !== null ) {
-                            curMetricValue = [parseFloat(curRegexResult[1])];
-                        }
-                    }
-
-                    curTestData["metrics"].push({name: curMetric, value: curMetricValue})
-                }
-
+                    
+                    curMetricValue = [];
+                    /*	Parse all values for single metric from result to an array
+                     *  e.g Liberty Startup =>
+                     *  curRegexResult = ['startup time in ms 32',32,'startup time in ms 41',41, x 6] 
+                     */
+                    curRegexResult = curSearchString.split(curBench['metrics'][i].regex);
+
+                    //collect only the capture groups from the regex array
+                    curMetricValue = curRegexResult.filter ( (value,index) => (index %2 == 1) ? value:null)
+   
+	                //check if any values 0. If there is, raise an error flag
+                    isValid = curMetricValue.some ( value => (value != 0))
```

Comment:
We shouldn't do this since some metric's value could actually be 0 in some rare case. Moreover, this would set it `testResult` to `FAILED`, which we shouldn't.